syft/serde.py

```diff
@@ -405,6 +405,46 @@ def _detail_ndarray(arr_representation: Tuple[bin, List[int], str]) -> numpy.nda
 
     return res
 
+#   slice
+
+
+def _simplify_slice(my_slice: slice) -> Tuple[int, int, int]:
```

Comment:
This function does not return a tuple. It returns a list

Reply:
I copied that one from line 317 

`def _simplify_range(my_range: range) -> Tuple[int, int, int]:`

Probably should adjust that line as well. 

```suggestion
def _simplify_slice(my_slice: slice) -> List:
```

Reply:
That's the same discussion @LaRiffle and I had. It returns a PYTHON List but the data structure is actually a tuple because it has exactly 3 elements. you don't want to get a list with 4 elements but a List with 3 integers in it.

That's why the return type is described as a Tuple of ints

Though we actually were not sure if the type annotations should describe the actual python return type or the actual data structure that is returned. I prefer data structure so you can catch if someone is returning nonsense.

Your thoughts? @robert-wagner @Ogofo

Reply:
Theoretically it should be both. If we don't have it correspond to the actual python return type, then we loose any potential benefit of automated type checking. But if we don't have it correspond to the theoretical data structure then we don't get help understanding what is actually going on. We could change this function to return a tuple pretty easily which would satisfy both

Reply:
Theoretically it should be both. If we don't have it correspond to the actual python return type, then we loose any potential benefit of automated type checking. But if we don't have it correspond to the theoretical data structure then we don't get help understanding what is actually going on. We could change this function to return a tuple pretty easily which would satisfy both

Reply:
Sure we can return a tuple. But we have to be aware that `msgpack`  is binarizing the tuple as a list. So we have to change the input `def _detail_slice(my_slice: Tuple[int, int, int]) -> slice:`  to `def _detail_slice(my_slice: List) -> slice:`. Think we should make this clear within the docstring.